### ISSUE TITLE
Minor adjustment to Guidelines For Respectful Communication

### ISSUE THREAD

**Issue by hasufell** [haskellfoundation/haskellfoundation.github.io#463]
When reading https://github.com/todogroup/opencodeofconduct/blob/gh-pages/index.md I found that the wording is much better.

Compare:

> Be welcoming: We strive to be a community that welcomes and supports people of all backgrounds and identities. This includes, but is not limited to members of any race, ethnicity, culture, national origin, colour, immigration status, social and economic class, educational level, sex, sexual orientation, gender identity and expression, age, size, family status, political belief, religion, and mental and physical ability.

With the HF text:

> We do not tolerate any form of discriminatory language or behaviour towards any minority (for example age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation).

The problems I see with the HF text is:

* although not intentionally, it feels like it's saying "discriminatory language is ok against majorities"
* unlike the open code of conduct text, it specifically omits "political belief" (why?)

I say this, because I have repeatedly seen discriminatory language and toxic behavior in the Haskell community against e.g. other Haskellers based on their political belief.

I'm not sure anyone reads the guidelines that closely, but it's hard to point to a community document saying "we welcome all people, regardless of their political belief, as long as they follow our guidelines for respectful communication", because that is simply not what the text says.

----

So what I propose is a wording like:

> We do not tolerate any form of discriminatory language or behaviour, especially towards minorities, and strive to welcome people of all backgrounds and identities (for example age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, political belief, religion, or sexual identity and orientation).

**Comment by Kleidukos** [comment]
That's a good rephrasing, I agree with @andreabedini

**Comment by hasufell** [comment]
Right, given that it's not really an enforceable policy... it seems positive language is more appropriate.

**Comment by jmct** [comment]
Thanks for bringing this up. There's actually a board meeting this week.

I'll make sure this is on the agenda.

**Comment by tomjaguarpaw** [comment]
Thanks for flagging this up @hasufell. I like your suggestion.

As a point of information, I believe these guidelines were taken directly from the GHC Steering Committee guidelines, published here:

https://github.com/ghc-proposals/ghc-proposals/blob/master/GRC.rst

There's no reason the HF's guidelines couldn't diverge, of course, and no reason both bodies couldn't adopt the same change at the same time. It's worth discussing further.

**Comment by simonpj** [comment]
I like the change. "Be welcoming" is much better than "We do not tolerate..". 

Moreover I suggest that we move that "Be welcoming" paragraph to be the first bullet, not the last.

**Comment by maralorn** [comment]
>     * unlike the open code of conduct text, it specifically omits "political belief" (why?)

Assuming this is an honest question, I wanted to answer it: The normal reason to exclude "political belief" from a list like this is because certain political beliefs are in contradiction to the intended meaning of the rest of the paragraph. It could be construed to e.g. mean "Let’s be welcoming to people who want to exclude others based on race". Of course most "political beliefs" should indeed be respected in a diverse community, and a rule which expresses that seems reasonable.

**Comment by tomjaguarpaw** [comment]
> The normal reason to exclude "political belief" from a list like this is because certain political beliefs are in contradiction to the intended meaning of the rest of the paragraph

Would you say that also applies to "religion" (which is included in the list)?

**Comment by tomjaguarpaw** [comment]
Yes, I beg your pardon, "political belief" doesn't correspond to "religion", it corresponds to "religious belief". How would it sound to you to say we welcome people regardless of their "politics"[^1], since we cannot conclude from a persons politics that they necessarily hold a *particular* problematic political belief.

[^1]: perhaps we can find a better word

**Comment by maralorn** [comment]
I fear that a GitHub thread is bad medium to have a detailed discussion about this and I expect that we would arrive at a reasonable agreement if we had it.

If someone says "All people of Group X are stupid and shouldn’t be allowed to program Haskell." in a discussion in our community, that would violate the rules of respectful communication in my opinion. Clearly, it wouldn’t be okay to justify this with "But you said you respect religion and this is my religious belief." neither with "But you said you respect political beliefs." nor "this is what people with my nationality think" or "I only believe this because of my socio-economic status."

I just wanted to remark that this is the tension which sometimes leads to "political beliefs" being excluded from that list, because they are probably the easiest to abuse in this way. I don’t intend to bikeshed the wording and as long as we all interpret the rules in good faith I don’t think this will matter.

**Comment by hasufell** [comment]
> We cannot conclude from a persons religion that they necessarily hold a certain problematic political belief.

I'd argue that's false. But I do not want to get into the details of that either.

I think the text as a whole already implies that we do not care about people's beliefs (religious, polticial or otherwise) as long as they follow the code of conduct.

That means it's fine if a persons belief is in tension with the code of conduct, but they adhere to it anyway.

**Comment by simonpj** [comment]
> That means it's fine if a persons belief is in tension with the code of conduct, but they adhere to it anyway.

I think @hasufell has it right here.   The point is that we want to

- Welcome everyone
- Irrespective of their age, gender, weight, nationality, religion -- or (I would agree) politics
- Provided that they *behave* in a respectful and gracious manner toward others

The guidelines should not (and do not) seek to control what a person _believes_, nor to exclude anyone because of those (alleged) beliefs!  They simply ask that if you want to participate in our community you should _behave_ in a respectful way.

I have not gone back to the doc to see how to better express this point, but I'm with @maralorn: we should not consume a lot of bandwidth on the details.  (For example, it's not clear to me that it's helpful to give a long list of characteristics, and then argue about what should be listed or not-listed.)  Rather let's make sure that the _intent_ (which I have tried to articulate above) is clear, and interpret those principles in good faith.   

Back to @hasufell's original point: I agree that "We welcome.." is better than "We do not tolerate...".

**Comment by hasufell** [comment]
I changed the wording to be a) positive and b) include "political belief": https://github.com/haskellfoundation/haskellfoundation.github.io/pull/480

I'm not sure that's too implicit. I would like a way to express what Simon correctly re-iterated: we care about peoples *behavior*, not their *beliefs*. But the guideline shouldn't sound like an abstract philosophical text.

So if anyone has an idea to communicate that more explicitly, please share your thoughts.
